plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1530,6 +1530,12 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         IcebergColumnHandle handle = (IcebergColumnHandle) column;
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
         try {
+            icebergTable.spec().fields().stream()
+                    .filter(field -> field.name().equals(handle.getName()))
```

Comment:
Filtering by name doesn't work well if the partition definition contains transformation. e.g. `day(ts)`